src/main/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListener.java

```diff
@@ -196,6 +197,26 @@ public final void onCompleted(final Run run, @Nonnull final TaskListener listene
           statsd = getDescriptor().leaseClient();
           statsd.incrementCounter("completed", tagsToCounter);
           logger.fine(String.format("Attempted to send 'completed' counter with tags: %s", Arrays.toString(tagsToCounter)));
+
+          // Send KPIs 
+          if (!isMasterBranch(run)) {
```

Comment:
Why do we only want to collect metrics for `!isMasterBranch(run)`?

Reply:
These metrics make only sense when it's about a release. And my assumption is that releases happens only on the master branch. That's why it doesn't make sense to consider PR or branch builds.

Reply:
Yea, I see how this is tied to code releases. We could hit issues trying to match a hardcoded `master` branch that could really limits these metrics. It's possible releases happen on branches not named `master` or some other use case with a name different from `master`. I think we'll need some options in the Job configuration page to make this flexible - a checkbox to enable/disable collection of these metrics as well as an option to specify a branch name. What are your thoughts?

Reply:
How about instead of making the branch a configuration value, tracking _all_ branches, and adding that as a tag, so that the user can then decide which ones they want to use for overlays, metrics, etc?

Reply:
💯 @miketheman

Reply:
If the branch should be added as tag, it would affect also all other metrics. Do you think that make sense?

Reply:
Maybe the branch tag should be associated with every metric, so as to allow for better analysis

Reply:
Apologies for not catching this earlier, but we already collect the branch tag if it exists: https://github.com/DataDog/jenkins-datadog-plugin/blob/27fc3a58014aa4a84a9c35173b7f8d254355c5e2/src/main/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListener.java#L314-L315

Reply:
Thanks for the hint. Actually, I've never seen this tag in one of our metrics. Have to check why this is missing.